test/lib/analyzer/image-inspector.spec.ts

```diff
@@ -182,6 +182,82 @@ describe("getImageArchive", () => {
     });
   });
 
+  describe("from remote registry with no platform feature binary", () => {
+    const customPath = "./new_custom/image/save/path";
+
+    afterEach(() => {
+      rmdirRecursive(customPath.split(path.sep));
+    });
+
+    it("should produce the expected state", async () => {
+      const imageSavePath = path.join(customPath, uuidv4());
+      // we simulate the Docker CLI being so old that the `--platform` flag is not supported at all.
+      const dockerPullCliSpy = jest
+        .spyOn(Docker.prototype, "pullCli")
+        .mockImplementation(() => {
+          return new Promise<subProcess.CmdOutput>((_, reject) => {
+            reject({
+              stdout: "",
+              stderr: "unknown flag: --platform\nSee 'docker pull --help'.",
+            });
+          });
+        });
+
+      await expect(
+        imageInspector.getImageArchive(targetImage, imageSavePath),
+      ).rejects.toEqual(
+        new Error(
+          '"--platform" is only supported on a Docker daemon with version later than 17.09',
+        ),
+      );
+
+      expect(dockerPullCliSpy).toHaveBeenCalled();
+
+      const imageExistsOnDisk: boolean = fs.existsSync(
+        path.join(imageSavePath, "image.tar"),
+      );
+      expect(imageExistsOnDisk).toBe(false);
+    });
+  });
+
+  describe("from remote registry with experimental platform feature binary", () => {
```

Comment:
This and the above new tests are pretty much the exact same apart from the returned errors.
Should I use a loop instead for this? 
Or only test one of these?